### PR TITLE
feat: udpate MosaicViewProperties

### PR DESCRIPTION
**Pull request by TCL735**
Closes https://github.com/influxdata/ui/issues/510

Updates the Mosaic View Properties to include hover dimension and two new properties for the y-axis. This fleshes out Mosaic graphs to render properly any data set. Also improves the hovering.

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [ ] Tests pass
- [x] http/swagger.yml updated (if modified Go structs or API)
- [x] Feature flagged (if modified API)

### CODE REVIEW

File: pkger/clone_resource.go [hunk 1]
```diff
@@ -832,11 +832,14 @@ func convertCellView(cell influxdb.Cell) chart {
 		ch.Kind = chartKindMosaic
 		ch.Queries = convertQueries(p.Queries)
 		ch.Colors = stringsToColors(p.ViewColors)
+		ch.HoverDimension = p.HoverDimension
 		ch.XCol = p.XColumn
 		ch.GenerateXAxisTicks = p.GenerateXAxisTicks
 		ch.XTotalTicks = p.XTotalTicks
 		ch.XTickStart = p.XTickStart
 		ch.XTickStep = p.XTickStep
+		ch.YLabelColumnSeparator = p.YLabelColumnSeparator
```
Comment (danxmoran) [inline pkger/clone_resource.go:841]: Should we do anything special here if `YLabelColumnSeparator` is empty, for backwards-compatibility?

File: pkger/clone_resource.go [hunk 2]
```diff
@@ -832,11 +832,14 @@ func convertCellView(cell influxdb.Cell) chart {
...
+		ch.YLabelColumns = p.YLabelColumns
```
Comment (danxmoran) [inline pkger/clone_resource.go:842]: Same question, should we do anything special for empty?

File: http/swagger.yml [hunk 3]
```diff
@@ -9271,6 +9277,9 @@ components:
           type: string
         ySuffix:
           type: string
+        hoverDimension:
+          type: string
+          enum: [auto, x, y, xy]
```
Comment (danxmoran) [inline http/swagger.yml:9282]: Should we mark one of these as the `default`?

File: pkger/clone_resource.go [hunk 4]
```diff
@@ -832,11 +832,14 @@ func convertCellView(cell influxdb.Cell) chart {
 		ch.Kind = chartKindMosaic
 		ch.Queries = convertQueries(p.Queries)
 		ch.Colors = stringsToColors(p.ViewColors)
+		ch.HoverDimension = p.HoverDimension
```
Comment (danxmoran) [inline pkger/clone_resource.go:835]: And here, should we handle empty / unset?

File: pkger/parser_models.go [hunk 5]
```diff
@@ -720,11 +724,14 @@ func (c *chart) properties() influxdb.ViewProperties {
...
 			Queries:                    c.Queries.influxDashQueries(),
 			ViewColors:                 c.Colors.strings(),
+			HoverDimension:             c.HoverDimension,
 			XColumn:                    c.XCol,
 			GenerateXAxisTicks:         c.GenerateXAxisTicks,
 			XTotalTicks:                c.XTotalTicks,
 			XTickStart:                 c.XTickStart,
 			XTickStep:                  c.XTickStep,
+			YLabelColumnSeparator:      c.YLabelColumnSeparator,
+			YLabelColumns:              c.YLabelColumns,
```
Comment (danxmoran) [inline pkger/parser_models.go:734]: Same questions about handling empty / unset for these new fields.

File: pkger/testdata/dashboard_mosaic.yml [hunk 6]
```diff
@@ -18,13 +18,19 @@ spec:
...
       yCol: _value
       width:  6
       height: 3
-      ySeriesColumns: ["_value", "foo"]
+      yLabelColumnSeparator: ","
+      yLabelColumns:
+        - foo
+      ySeriesColumns:
+        - _value
+        - foo
```
Comment (danxmoran) [inline pkger/testdata/dashboard_mosaic.yml:33]: If you leave the test and this file unchanged, does the existing test logic still pass?

**Review by danxmoran — COMMENTED** [review]
(Caveat: I'm not as familiar with this area of the code)

My main concern is around backwards-compatibility with existing models that don't specify these fields.

Also: could you add a line to `CHANGELOG.md`?

File: http/swagger.yml (hunk 3 above)
Comment (TCL735) [reply http/swagger.yml:9282]: No. Empty string will be interpreted correctly by the UI.

File: pkger/clone_resource.go (hunk 4 above)
Comment (TCL735) [reply pkger/clone_resource.go:835]: No need.

File: pkger/clone_resource.go (hunk 1 above)
Comment (TCL735) [reply pkger/clone_resource.go:841]: No need. Empty string will be interpreted correctly by the UI.

File: pkger/clone_resource.go (hunk 2 above)
Comment (TCL735) [reply pkger/clone_resource.go:842]: No. Empty array will be interpreted correctly by the UI.

File: pkger/parser_models.go (hunk 5 above)
Comment (TCL735) [reply pkger/parser_models.go:734]: No need. They will be interpreted correctly.

File: pkger/testdata/dashboard_mosaic.yml (hunk 6 above)
Comment (TCL735) [reply pkger/testdata/dashboard_mosaic.yml:33]: Yes.

**Comment by TCL735** [comment]
Changelog updated.

**Review by danxmoran — APPROVED** [review]

**Comment by TCL735** [comment]
The latest force push was to rebase on top of master.

**Pull request merged**